### ISSUE TITLE
Floating point exception when invoking on crostini / Chrome M99 on Acer C933 Chromebook

### ISSUE THREAD

**Issue by textures2** [open-mpi/hwloc#525]
### What version of hwloc are you using?

* Run `which lstopo` to ensure that you are running the desired hwloc installation

```/usr/bin/lstopo
```

* Run `lstopo --version` to find their hwloc version

2.4.1

### Which operating system and hardware are you running on?

`Linux penguin 5.10.92-14532-g179c52887ab5 #1 SMP PREEMPT Tue Mar 1 18:42:32 PST 2022 x86_64 GNU/Linux`

* On Unix-like systems, run `uname -a` so that we know which operating system, distribution, and kernel version you are using.
* Post the output of `lstopo -` if it works

Does not work :(

* Describe the machine and the processors it contains, as well as any memory and/or peripherals that matter to your issue

As mentioned above this is on a Chromebook in a linux environment.

### Details of the problem
This issue arises from
https://github.com/xmrig/xmrig/issues/2956#issuecomment-1061538209

**Comment by bgoglin** [comment]
Hello.
Setting HWLOC_COMPONENTS=-x86 should workaround the issue by disabling the x86 backend entirely.
Can you set this variable and run "hwloc-gather-cpuid cpuid"? It will create a "cpuid" directory that you can send for debugging.

**Comment by textures2** [comment]
The device has an x86 cpu (N4000 I think?), though, will "disabling the x86
backend" still allow it to work?

**Comment by bgoglin** [comment]
Thanks, I can reproduce here. It looks Intel didn't properly implement all CPUID leaves in this celeron processor. I'll work on a workaround.

**Comment by textures2** [comment]
Thanks for the awesomely fast reply.  The part I cannot understand is why updating versions of chromeos would have caused this to change.

**Comment by bgoglin** [comment]
Is ChromeOS implemented with virtual machines? VMs may have to intercept the CPUID instructions to return virtual info about virtual CPUs instead of real info about physical CPUs. If they don't intercept/reimplement those instructions properly, boom.

**Comment by textures2** [comment]
Yes it is [Crostini](url).  Maybe a bug needs to be opened with them also?

**Comment by bgoglin** [comment]
The issue is:
CPUID leaf 0x1 returns register EDX with bit HTT=28 unset. According to the Intel x86 manual, this means: "A value of 0 for HTT indicates there is only a single logical processor in the package and software should assume only a single APIC ID is reserved."
This seems wrong on a quad-core celeron 4100 processor.

Moreover, CPUID leaf 0x4 with subleaf 0 returns a valid first level of cache which says (in EAX bits 26-31)that there are 4 cores in the physical package. This is correct, and contradicts CPUID leaf 0x1 above.

**Comment by bgoglin** [comment]
I pushed the fix to git master and v2.7. It will be released in the upcoming hwloc 2.7.1. In the meantime, either keep the environment variable, or use a hwloc nightly snapshot such as a tarball at https://ci.inria.fr/hwloc/job/basic/job/v2.7/

If you report the Crostini issue, please let me know. Or I can do it if you give me more details about your setup, for instance which version of it you had before and after the upgrade broke thins (I don't even know what Crostini is and where/how it is developed).

**Comment by textures2** [comment]
Opened an issue with Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1304418

**Comment by bgoglin** [comment]
Thanks, you should point them to the details I gave at https://github.com/open-mpi/hwloc/issues/525#issuecomment-1061640610

**Comment by textures2** [comment]
I did just that, thanks.
